salt/utils/x509.py

```diff
@@ -0,0 +1,2089 @@
+    "nameConstraints": ("X509v3 Name Constraints",),
+    "noCheck": ("OCSP No Check",),
+    "tlsfeature": ("TLS Feature",),
+    "nsComment": ("Netscape Comment",),
+    "nsCertType": ("Netscape Certificate Type",),
+    "cRLNumber": ("X509v3 CRLNumber",),
+    "deltaCRLIndicator": ("X509v3 Delta CRL Indicator",),
+    "freshestCRL": ("x509v3 Freshest CRL",),
+}
+
+EXTENSIONS_OID = {
+    "basicConstraints": cx509.ExtensionOID.BASIC_CONSTRAINTS,
+    "keyUsage": cx509.ExtensionOID.KEY_USAGE,
+    "extendedKeyUsage": cx509.ExtensionOID.EXTENDED_KEY_USAGE,
+    "subjectKeyIdentifier": cx509.ExtensionOID.SUBJECT_KEY_IDENTIFIER,
+    "authorityKeyIdentifier": cx509.ExtensionOID.AUTHORITY_KEY_IDENTIFIER,
+    "issuerAltName": cx509.ExtensionOID.ISSUER_ALTERNATIVE_NAME,
+    "authorityInfoAccess": cx509.ExtensionOID.AUTHORITY_INFORMATION_ACCESS,
+    "subjectAltName": cx509.ExtensionOID.SUBJECT_ALTERNATIVE_NAME,
+    "crlDistributionPoints": cx509.ExtensionOID.CRL_DISTRIBUTION_POINTS,
+    "issuingDistributionPoint": cx509.ExtensionOID.ISSUING_DISTRIBUTION_POINT,
+    "certificatePolicies": cx509.ExtensionOID.CERTIFICATE_POLICIES,
+    "policyConstraints": cx509.ExtensionOID.POLICY_CONSTRAINTS,
+    "inhibitAnyPolicy": cx509.ExtensionOID.INHIBIT_ANY_POLICY,
+    "nameConstraints": cx509.ExtensionOID.NAME_CONSTRAINTS,
+    "noCheck": cx509.ExtensionOID.OCSP_NO_CHECK,
+    "tlsfeature": cx509.ExtensionOID.TLS_FEATURE,
+    "nsComment": None,
+    "nsCertType": None,
+    "cRLNumber": cx509.ExtensionOID.CRL_NUMBER,
+    "deltaCRLIndicator": cx509.ExtensionOID.DELTA_CRL_INDICATOR,
+    "freshestCRL": cx509.ExtensionOID.FRESHEST_CRL,
+}
+
+EXTENSIONS_CRL_ENTRY_OID = {
+    "certificateIssuer": cx509.CRLEntryExtensionOID.CERTIFICATE_ISSUER,
+    "CRLReason": cx509.CRLEntryExtensionOID.CRL_REASON,
+    "invalidityDate": cx509.CRLEntryExtensionOID.INVALIDITY_DATE,
+}
+
+
+EXTENDED_KEY_USAGE_OID = {
+    "serverAuth": cx509.ObjectIdentifier("1.3.6.1.5.5.7.3.1"),
+    "clientAuth": cx509.ObjectIdentifier("1.3.6.1.5.5.7.3.2"),
+    "codeSigning": cx509.ObjectIdentifier("1.3.6.1.5.5.7.3.3"),
+    "emailProtection": cx509.ObjectIdentifier("1.3.6.1.5.5.7.3.4"),
+    "timeStamping": cx509.ObjectIdentifier("1.3.6.1.5.5.7.3.8"),
+    "OCSPSigning": cx509.ObjectIdentifier("1.3.6.1.5.5.7.3.9"),
+    "msSmartcardLogin": cx509.ObjectIdentifier("1.3.6.1.4.1.311.20.2.2"),
+    "pkInitKDC": cx509.ObjectIdentifier("1.3.6.1.5.2.3.5"),
+    "ipsecIKE": cx509.ObjectIdentifier("1.3.6.1.5.5.7.3.17"),
+    "msCodeInd": cx509.ObjectIdentifier("1.3.6.1.4.1.311.2.1.21"),
+    "msCodeCom": cx509.ObjectIdentifier("1.3.6.1.4.1.311.2.1.22"),
+    "msCTLSign": cx509.ObjectIdentifier("1.3.6.1.4.1.311.10.3.1"),
+    "msEFS": cx509.ObjectIdentifier("1.3.6.1.4.1.311.10.3.4"),
+}
+
+ACCESS_OID = {
+    "OCSP": cx509.AuthorityInformationAccessOID.OCSP,
+    "caIssuers": cx509.AuthorityInformationAccessOID.CA_ISSUERS,
+    "caRepository": SubjectInformationAccessOID.CA_REPOSITORY,
+}
```

Comment:
These global dictionaries should be [frozen](https://github.com/saltstack/salt/blob/v3005.1/salt/utils/immutabletypes.py#L111)